address-space-controller/src/main/java/io/enmasse/controller/RouterConfigController.java

```diff
@@ -0,0 +1,377 @@
+            router.setWorkerThreads(routerSpec.getWorkerThreads());
+        }
+
+        // SSL Profiles
+        SslProfile authServiceSsl = new SslProfile();
+        authServiceSsl.setName("auth_service_ssl");
+        authServiceSsl.setCaCertFile("/etc/qpid-dispatch/authservice-ca/tls.crt");
+
+        SslProfile sslDetails = new SslProfile();
+        sslDetails.setName("ssl_details");
+        sslDetails.setCertFile("/etc/qpid-dispatch/ssl/tls.crt");
+        sslDetails.setPrivateKeyFile("/etc/qpid-dispatch/ssl/tls.key");
+
+        SslProfile interRouterTls = new SslProfile();
+        interRouterTls.setName("inter_router_tls");
+        interRouterTls.setCaCertFile("/etc/enmasse-certs/ca.crt");
+        interRouterTls.setCertFile("/etc/enmasse-certs/tls.crt");
+        interRouterTls.setPrivateKeyFile("/etc/enmasse-certs/tls.key");
+
+        // Authenticationservice plugin
+        AuthServicePlugin authService = new AuthServicePlugin();
+        authService.setName("auth_service");
+        authService.setHost(authServiceSettings.getHost());
+        authService.setPort(authServiceSettings.getPort());
+        authService.setRealm(authServiceSettings.getRealm());
+        authService.setSslProfile("auth_service_ssl");
+
+        // Listeners
+        Listener localBypass = new Listener();
+        localBypass.setHost("127.0.0.1");
+        localBypass.setPort(7777);
+        localBypass.setAuthenticatePeer(false);
+
+        Listener livenessProbe = new Listener();
+        livenessProbe.setHost("127.0.0.1");
+        livenessProbe.setPort(7770);
+        livenessProbe.setAuthenticatePeer(false);
+        livenessProbe.setHttp(true);
+        livenessProbe.setMetrics(false);
+        livenessProbe.setHealthz(true);
+        livenessProbe.setWebsockets(false);
+        livenessProbe.setHttpRootDir("invalid");
+
+        Listener interRouter = new Listener();
+        interRouter.setHost("0.0.0.0");
+        interRouter.setPort(55672);
+        interRouter.setRole(Role.inter_router);
+        interRouter.setAuthenticatePeer(true);
+        interRouter.setSslProfile("inter_router_tls");
+        interRouter.setSaslMechanisms("EXTERNAL");
+        if (routerSpec != null) {
+            if (routerSpec.getIdleTimeout() != null) {
+                interRouter.setIdleTimeoutSeconds(routerSpec.getIdleTimeout());
+            }
+
+            if (routerSpec.getLinkCapacity() != null) {
+                interRouter.setLinkCapacity(routerSpec.getLinkCapacity());
+            }
+        }
+
+        Listener httpsPublic = new Listener();
+        httpsPublic.setHost("0.0.0.0");
+        httpsPublic.setPort(8443);
+        httpsPublic.setSaslPlugin("auth_service");
+        httpsPublic.setSslProfile("ssl_details");
+        httpsPublic.setHttp(true);
+        httpsPublic.setAuthenticatePeer(true);
+        if (routerSpec != null && routerSpec.getPolicy() != null) {
+            httpsPublic.setPolicyVhost("public");
```

Comment:
I notice the io.enmasse.controller.common.TemplateParameter now has unused constants.